CMakeLists.txt

```diff
@@ -135,24 +147,35 @@ add_subdirectory(drivers)
 add_subdirectory(hal)
 add_subdirectory(platform)
 add_subdirectory(rtos)
-add_subdirectory(targets)
+
+if(${CMAKE_CROSSCOMPILING})
+    add_subdirectory(targets)
+    # The directories below contain optional target libraries
+    add_subdirectory(connectivity EXCLUDE_FROM_ALL)
```

Comment:
Yes most of the components have unit test and we want to include to the build when we invoke unit test cmake command but in case of application build (Cross compiling), we would like to exclude to the build by default.  When the application cmake target_link_libraries to any of the excluded component then it gets added into the build.

Reply:
I see why but this has implications beyond unit tests. I would suggest the specific unit testing targets are excluded from the `all` target in the appropriate places, not at the top level here.

Reply:
I see why but this has implications beyond unit tests. I would suggest the specific unit testing targets are excluded from the `all` target in the appropriate places, not at the top level here.

Reply:
If we remove `EXCLUDE_FROM_ALL` then all the added CMake targets will be compiled when we try to build the CMake `all` target (mbed-tools tries to build the all target as it has no knowledge of CMake target names). Using `EXCLUDE_FROM_ALL` here allows us to specify which components of Mbed are built by using target_link_libraries in the consuming application. This isn't an ideal set-up but it was a pre-existing issue. I'd suggest we add an issue to the CMake milestone to tidy up here.

Reply:
> I'd suggest we add an issue to the CMake milestone to tidy up here.

@rwalton-arm can you create an issue please for this?